ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1297,7 +1303,12 @@ public void renamePartition(Table tbl, Map<String, String> oldPartSpec, Partitio
     }
   }
 
-  // TODO: this whole path won't work with catalogs
+  /**
+   * When you call this method, you need to ensure that the catalog has been set in the db object.
```

Comment:
why not have fallback similar to https://github.com/apache/hive/pull/6088/files#diff-9d6bc2f1a2972d844b3107f29bc1731af7adf6f4c922e08904345c5ae88cd727R1443